Core/Core/SwiftUIViews/ListCellView.swift

```diff
@@ -0,0 +1,217 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+struct ListCellView: View {
+
+    enum ListCellStyle {
+        case mainAccordionHeader
+        case listAccordionHeader
+        case listItem
+    }
+
+    enum SelectionState {
+        case deselected
+        case selected
+        case partiallySelected
+    }
+
+    let cellStyle: ListCellStyle
+    let title: String
+    let subtitle: String?
+    let selectionState: SelectionState
+    let isCollapsed: Bool?
+    let selectionDidToggle: (() -> Void)?
+    let collapseDidToggle: (() -> Void)?
+
+    private var backgroundColor: Color {
+        switch cellStyle {
+        case .mainAccordionHeader:
+            return .backgroundLight
+        default:
+            return .backgroundLightest
+        }
+    }
+
+    private var cellHeight: CGFloat {
+        switch cellStyle {
+        case .mainAccordionHeader:
+            return 72.0
+        default:
+            return 54.0
+        }
+    }
+
+    @ViewBuilder
+    private var iconImage: some View {
+        HStack {
+            switch selectionState {
+            case .deselected:
+                Image("emptyLine", bundle: .core)
+                    .size(20)
+                    .foregroundColor(.textDarkest)
+            case .selected:
+                Image("completeSolid", bundle: .core)
+                    .size(20)
+                    .foregroundColor(.textInfo)
+            case .partiallySelected:
+                Image("partialSolid", bundle: .core)
```

Comment:
`Image.partialSolid` should be available after you run the generate icons script.